torchrl/data/tensordict/tensordict.py

```diff
@@ -2195,17 +2197,14 @@ def _stack_onto_at_(
     def set_at_(
         self, key: str, value: Union[dict, COMPATIBLE_TYPES], idx: INDEX_TYPING
     ) -> TensorDictBase:
-        if isinstance(value, dict):
-            value = TensorDict(
-                value, batch_size=self.batch_size, device=self.device_safe()
-            )
         if not isinstance(key, str):
             raise TypeError(f"Expected key to be a string but found {type(key)}")
 
         # do we need this?
-        # value = self._process_tensor(
-        #     value, check_tensor_shape=False, check_device=False
-        # )
+        if not isinstance(value, _accepted_classes):
```

Comment:
For record: I guess we want to process the input only if it's not of the right class, and not check the device

I think the previous logic was: if you're calling set_at_ you're basically doing something like `tensor.copy_(smth)`. The only requirement of tensor.copy is "The src tensor must be [broadcastable](https://pytorch.org/docs/stable/notes/broadcasting.html#broadcasting-semantics) with the self tensor. It may be of a different data type or reside on a different device." (see [here](https://pytorch.org/docs/stable/generated/torch.Tensor.copy_.html)) so I guess we're fine with this